### PR TITLE
feat(compiler): parse5 DOM adapter

### PR DESCRIPTION
**Pull request by marclaval**
Turning experiments into an adapter for NodeJS, following @yjbanov model.

Needed for #760

### CODE REVIEW

File: modules/angular2/src/dom/parse5_adapter.cjs [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+  clone(node) {
+    var temp = treeAdapter.createElement("template", null, []);
+    treeAdapter.appendChild(temp, node);
+    var serialized = serializer.serialize(temp);
+    var newParser = new parse5.Parser(parse5.TreeAdapters.htmlparser2);
+    return newParser.parseFragment(serialized).childNodes[0];
+  }
+  hasProperty(element, name:string) {
+    //TODO: build a proper list, this one is all the keys of a HTMLInputElement
+    var propertyList = ["webkitEntries","incremental","webkitdirectory","selectionDirection","selectionEnd",
```
Comment (yjbanov) [inline modules/angular2/src/dom/parse5_adapter.cjs]: we should probably avoid building this list each time `hasProperty` is called

File: modules/angular2/src/dom/parse5_adapter.cjs [hunk 2]
```diff
@@ -0,0 +1,333 @@
+var parse5 = require('parse5');
```
Comment (yjbanov) [inline modules/angular2/src/dom/parse5_adapter.cjs]: This adapter is very feature-complete. However, only a fraction of it is exercised by the compiler and compiler tests. I'm not sure we want to keep unused and untested code around.

File: gulpfile.js [hunk 3]
```diff
@@ -594,6 +594,10 @@ gulp.task('test.unit.dart/ci', function (done) {
   karma.start({configFile: __dirname + '/karma-dart.conf.js',
       singleRun: true, reporters: ['dots'], browsers: getBrowsersFromCLI()}, done);
 });
+gulp.task('test.unit.cjs/ci', function (done) {
+  return gulp.src(CONFIG.dest.js.cjs + '/angular2/test/unittests.server.spec.js').pipe(jasmine());
```
Comment (yjbanov) [inline gulpfile.js]: Could you do the same thing we did for Dart? I.e. glob all `*.server.spec.js` files and run them.

File: modules/angular2/src/dom/parse5_adapter.cjs (hunk 2 above)
Comment (mlaval) [inline modules/angular2/src/dom/parse5_adapter.cjs]: It aims at running all compiler tests in NodeJS, but this is still not completed. Incremental commits would ease the task.

**Comment by mlaval** [comment]
Coming back with a more complete adapter. With it, it is possible to successfully run all the tests of the compiler in NodeJS.
In fact, 361 tests are run, and 5 are disabled: 4 because they rely on DOM events, 1 because of an issue in the CSS parser.

Please only review the last commit, the other 3 are pre-requisites which are covered by other PR.

**Comment by mhevery** [comment]
LGTM, @yjbanov can you give the final approval

**Pull request closed without merging**